scripts/overton/EDA_script.py

```diff
@@ -0,0 +1,176 @@
+import sys  
+
+import pandas as pd
+from collections import Counter
+import numpy as np
+from wordcloud import WordCloud
+import matplotlib.pyplot as plt
+import spacy
+import seaborn as sns
+import typer
+
+#EDA_FOR_NLP_PATH = 'C:/Users/Hubert/Dropbox/DarlingProject/eda_for_nlp_package'
+#FINAL_TABLE_PATH = 'C:/Users/Hubert/Documents/DarlingProject/Overton/EDA/final_table.parquet'
+#SPACY_TABLE_PATH = 'C:/Users/Hubert/Documents/DarlingProject/Overton/EDA/spacy_table_new.parquet'
+
+def main(eda_for_nlp_path : str, final_table_path : str, spacy_table_path : str):
+    sys.path.insert(0, eda_for_nlp_path)
+    for p in sys.path:
+        print( p )
+    import eda_for_nlp_package as eda
+    en = spacy.load("en_core_web_md")
+
+    df = pd.read_parquet(final_table_path)
+    df = df[['Title','Text','n_paragraphs','n_words']]
+    spacy_df = pd.read_parquet(spacy_table_path)
+
+    # create report
+    with open("EDA_report.txt", "w") as file_object:
+        file_object.write("This is report for string values. \n")
+
+    Text_all = []
+    Text_paragraph = []
+    for i in range(len(df)):
+        if df['Text'][i] is not None:
+            Text_all.append(df['Text'][i].sum())
+            for j in range(len(df['Text'][i])):
+                Text_paragraph.append(df['Text'][i][j])
+    else:
+        Text_all.append("")
+    df['Text_all'] = Text_all
+    with open("EDA_report.txt", "a") as file_object:
+        file_object.write("Number of paragraphs:" + str(len(Text_paragraph)) + "\n")
+    len(Text_paragraph)
+
+    titles = []
+    n_ai = []
+    n_ml = []
+    n_words = []
+    ai_fq = []
+    ml_fq = []
+    for i in range(len(df)):
+        titles.append(df['Title'][i])
+        n_ai.append(df['Text_all'][i].count('artificial intelligence'))
+        n_ml.append(df['Text_all'][i].count('machine learning'))
+        n_words.append(len(df['Text_all'][i]))
+        ai_fq.append(n_ai[i] / (n_words[i] + 1) * 100)
+        ml_fq.append(n_ml[i] / (n_words[i] + 1) * 100)
+    stats = pd.DataFrame({"title": titles, "n_ai": n_ai, "n_ml": n_ml, "n_words": n_words, "ai_fq(%)": ai_fq, "ml_fq(%)": ml_fq})
+    stats.to_excel('excel/stats.xlsx')

```

Comment:
czemu zapisujemy do excela?

Reply:
Podobnie jak w drugim skrypcie, żeby było łatwe do odczytania dla człowieka, ale tutaj akurat mogę zmienić na parquet, bo nie miałem pewności które wybrać

Reply:
Ale xlsx wg mnie w sumie jest spoko, bo tutaj nie ma dużych danych w statystykach (to nie tysiące słów). Daj znać które wolisz

Reply:
Ale xlsx wg mnie w sumie jest spoko, bo tutaj nie ma dużych danych w statystykach (to nie tysiące słów). Daj znać które wolisz